apps/dcos_overlay/src/dcos_overlay_lashup_kv_listener.erl

```diff
@@ -136,3 +156,31 @@ to_str(Mac) when length(Mac) =:= 6 ->
     lists:flatten(string:join(List, ":"));
 to_str(IP) ->
     inet:ntoa(IP).
+
+%%%===================================================================
+%%% Metrics functions
+%%%===================================================================
+
+-spec(init_metrics() -> ok).
+init_metrics() ->
+    prometheus_counter:declare([
+        {registry, overlay},
+        {name, lashup_events_received_total},
+        {help, "Total number of Lashup events received."}]),
+    prometheus_counter:declare([
+        {registry, overlay},
+        {name, reconciliations_total},
+        {help, "Total number of overlay configuration reconciliations attempted."}]),
+    prometheus_counter:declare([
+        {registry, overlay},
+        {name, configurations_applied_total},
+        {help, "Total number of configurations applied."}]),
+    prometheus_counter:declare([
+        {registry, overlay},
+        {name, configuration_failures_total},
+        {help, "Total number of overlay configuration failures."}]),
+    prometheus_counter:declare([
+        {registry, overlay},
+        {name, configurations_timed_out_total},
```

Comment:
I'd merge it into `configuration_failures_total`, no need to differentiate these two cases in metrics.